scripts/src/analyzeMessages.ts

```diff
@@ -0,0 +1,133 @@
+import Path from "path";
+import { promises as fs } from "fs";
+import { MongoClient, Db } from "mongodb";
+import { Conversation, Message } from "chat-server";
+import { makeTypeChatJsonTranslateFunc, assertEnvVars } from "chat-core";
+import { MessageAnalysis } from "./MessageAnalysis";
+
+import "dotenv/config";
+
+const {
+  MONGODB_DATABASE_NAME,
+  MONGODB_CONNECTION_URI,
+  OPENAI_ENDPOINT,
+  OPENAI_API_KEY,
+  OPENAI_CHAT_COMPLETION_MODEL_VERSION,
+  OPENAI_CHAT_COMPLETION_DEPLOYMENT,
+} = assertEnvVars({
+  MONGODB_DATABASE_NAME: "",
+  MONGODB_CONNECTION_URI: "",
+  OPENAI_ENDPOINT: "",
+  OPENAI_API_KEY: "",
+  OPENAI_CHAT_COMPLETION_MODEL_VERSION: "",
+  OPENAI_CHAT_COMPLETION_DEPLOYMENT: "",
+});
+
+async function main() {
+  const client = await MongoClient.connect(MONGODB_CONNECTION_URI);
+  try {
+    const db = client.db(MONGODB_DATABASE_NAME);
+    await analyzeMessages({ db });
+  } finally {
+    await client.close();
+  }
+}
+
+main();
+
+const analyzeMessages = async ({ db }: { db: Db }) => {
+  const conversationsCollection = db.collection<Conversation>("conversations");
+  const originalMessages = conversationsCollection.aggregate<Message>([
+    {
+      // Find messages in a recent timeframe
+      $match: {
+        "messages.createdAt": {
+          // Originals for older conversations are likely to have been deleted,
+          // so we can't use it to get the topic
+          $gte: daysBeforeDate(8),
+        },
```

Comment:
```suggestion
        },
        "messages": {
          "$elemMatch": { role: "user" }
        }
```

We can add this so that our initial match only includes conversations that actually have a user message (~30% do not have a user message)